### PR TITLE
Plotters integration

### PR DESCRIPTION
**Pull request by 38**
Hi, just as we discussed under issue #313 , this PR and this is very WIP in a very early stage. Any comments are welcome :)

- [x] Refactor `criterion::plot` code,  using dynamic dispatch for generating plots in `Html` code.
    - [ ] Extract the common data processing logic to the `PlotContext` and `PlotData` struct and reuse them for all backends
- [x] `Plotters` backend implemnetaion
    - [x] Distributions
    - [x] PDFs
    - [x] T-Test
    - [x] Regressions
    - [x] Line Comp and Violin 
- [x] Commandline argument for switching backends

### CODE REVIEW

**Comment by bheisler** [comment]
Great! Thanks for helping out. I've skimmed the code changes and they look pretty reasonable so far.

I did notice that you're using a git dependency on `plotters` - that will need to be changed to a regular crate dependency before this is merged. Crates.io doesn't allow crates to have git dependencies.

**Comment by 38** [comment]
> Great! Thanks for helping out. I've skimmed the code changes and they look pretty reasonable so far.
> 
> I did notice that you're using a git dependency on `plotters` - that will need to be changed to a regular crate dependency before this is merged. Crates.io doesn't allow crates to have git dependencies.

Yep, there's a lot of things need to be clean up (But I am going to resolve them when it's done). 
One of the major reason for using latest version is I am also thinking about making change from Plotters side to make this PR more reasonable and maybe add essential features to Plotters as well.  Once I finished the PR ending up with new features added into Plotters, I think I can publish a new version of Plotters and make Criterion dependents on that version.

**Comment by 38** [comment]
Seems I just got the PR fulling functioning, but I think there's still few things needs to be cleanup. If you have any thoughts please leave comments. Thank you ! 

-----

As you mentioned before, you concerned about adding Plotter would introduce other dependencies that slow down the compilation. I believe it's fine to making plotters a required dependencies as Plotters won't introduce too much dependencies if only SVG backend is enabled.(The only required one is num-trait).

```
$ cargo tree -p plotters
plotters v0.2.11
├── num-traits v0.2.9
│   [build-dependencies]
│   └── autocfg v0.1.7
└── svg v0.6.0
```

svg is a pretty tiny crate, plus plotters itself doesn't takes too long time to compile, I believe the additional compile time should be less than 5s. And `num-traits` is a dependency of criterion already. 

-------

Just realized svg crate actually compiles pretty slow, so I just remove this dependency completely. Now I believe there's no significant slow down for introducing plotters deps.
image
Uploads the timing plot generated by cargo

File: Cargo.toml [hunk 1]
```diff
@@ -32,6 +32,12 @@ rand_core = { version = "0.5", default-features = false }
 rand = "0.7"
 rayon = "1.1"
 
+[dependencies.plotters]
+# This should be changed to Plotters' next release
```
Comment (38) [inline Cargo.toml]: TODO: Change this to next published version

**Comment by bheisler** [comment]
Hey, thanks for your patience. This all looks great! I'll probably do some minor tweaks after it's merged. Thanks for the (awesome!) pull request!

**Pull request merged**